features/index.rst

```diff
@@ -114,3 +118,68 @@ Because ``roundedPetalsSource`` is a combination of a source and a function, it
 
 Because NStack functions, source, and sinks can be composed and reused, this lets you build powerful abstractions over infrastructure.
 
+
+.. _features_versioning
+
+Versioning
+----------
+
+Modules in NStack are versioned with a 3-digit suffix that is intended to follow semantic versioning, e.g.::
+  
+  Demo:0.0.1
+  
+This is specified in the ``nstack.yaml`` for code-based modules, and in ``workflow.nml`` for workflow modules.
+A module of a specific version is completely immutable, and it's not possible to build another copy of the module with the same version without deleting it first.
+
+Snapshots
+^^^^^^^^^
+
+When creating a new module, i.e. with ``nstack init``, your module will have the version number (``0.0.1-SNAPSHOT``). 
+The ``SNAPSHOT`` tag tells NStack to allow you to override it every time you build. 
+This is helpful for development, as you do not need to constantly increase the version number. 
+When you deem your module is ready for release, you can remove ``SNAPSHOT`` and NStack will create an immutable version of ``0.0.1``.
+
+.. _features_configuration
+
+Configuration
+-------------
+
+In addition to receiving input at runtime, modules, sources, and sinks often need to be able to configured by a workflow author. To do this, we use brackets and pass in a list of named records: ::
+
+   Sources.Postgres<Text> {
+        pg_host = "localhost", 
+        pg_port = "5432",
+        pg_user = "user", 
+        pg_password = "123456",
+        pg_database = "db", 
+        pg_query = "SELECT * FROM tbl;" 
+    }
+
+For sources and sinks, some parameters are mandatory, and some provide sensible defaults. This is documented in `Supported Integrations <supported_integrations>`_.
+
+To pass configuration parameters to a module, we use the same syntax ::
+
+  FirstLastName.full_name { first_name = "John" }
+
+NStack passes in configuration parameters as a dictionary, ``args``, which is added to the base class of your module.
+For instance, in Python you can access configuration parameters in the following manner:
+
+.. code :: python 
+  
+  class Service(nstack.BaseService):
+  
+      def full_name(self, second_name):
+        full_name = "{} {}".format(self.args.get("first_name", "Tux"), second_name)
+        return full_name
+
+
+.. _features_framework
+
+
+Framework Modules
+-----------------
+
+You may want to create a common parent module that has lots of complex dependencies already installed, either to save time or for standardisation. NStack supports this with *Framework Modules*. Simply create a new module similar to above, ``nstack init framework [parent]``, and modify the resulting ``nstack.yaml`` as needed.
```

Comment:
"It is often useful to create a common parent module with dependencies already installed"